test/typings/test-d/postgres-json.test-d.ts

```diff
@@ -169,6 +169,39 @@ async function testPostgresJsonAgg(db: Kysely<Database>) {
       status: string | null
     }[]
   }>(r4)
+
+  const r5 = await db
+    .selectFrom('person')
+    .innerJoin('pet', 'pet.owner_id', 'person.id')
+    .select((eb) => ['first_name', eb.fn.jsonAgg('owner_id').as('pet_names')])
+    .groupBy('person.first_name')
+    .execute()
+
+  expectType<
+    {
+      first_name: string
+      pet_names: number[] | null
+    }[]
+  >(r5)
+
+  const r6 = await db
+    .selectFrom('person')
+    .select((eb) => [
+      'first_name',
+      eb
+        .selectFrom('pet')
+        .select((eb) => [eb.fn.jsonAgg('pet.owner_id').as('pet_names')])
+        .whereRef('pet.owner_id', '=', 'person.id')
+        .as('pet_names'),
+    ])
+    .execute()
+
+  expectType<
+    {
+      first_name: string
+      pet_names: number[] | null
+    }[]
+  >(r6)
```

Comment:
This one doesn't add much value, let's remove it.

```suggestion
```

Reply:
Could be that this typing-wise doesn't make a difference, but to me this is another use-case. It's using the json-agg in a subquery rather than in a joined table. In a joined table you need a `groupBy`, but in a subquery you don't.

Reply:
Typings tests are there to ensure we got things right on the type-level. Not to document the SQL spec.

The relationship between a root query and it's sub-queries has been thoroughly tested elsewhere.
Group by doesn't affect anything on the type-level. This test case doesn't add any value here.

Excessive tests means:

1. more stuff to maintain.
2. more stuff to wait for locally or in CI.